protobuf/lib/src/protobuf/extension_registry.dart

```diff
@@ -134,7 +134,7 @@ T _reparseMessage<T extends GeneratedMessage>(
         resultMap ??= ensureResult()._fieldSet._values[field.index!];
 
     if (field.isRepeated) {
-      final messageEntries = message._fieldSet._values[field.index!];
+      PbList? messageEntries = message._fieldSet._values[field.index!];
```

Comment:
Same here -- it seems like we could avoid type casts in AOT and JIT when the value is `null`.